### PR TITLE
Cleanup image manipulator

### PR DESCRIPTION
**Pull request by allouis**
This cleans up the image manipulator library to reuse code where possible - including errors :+1:

The biggest change is refactoring the process method to use the imageResize method so we're not duplicating logic.

### CODE REVIEW

File: core/server/lib/image/manipulator.js [hunk 1]
```diff
@@ -8,60 +8,21 @@ const fs = require('fs-extra');
...
-        return Promise.reject(new common.errors.InternalServerError({
-            message: 'Sharp wasn\'t installed',
-            code: 'SHARP_INSTALLATION',
-            err: err
-        }));
-    }
-
-    // @NOTE: workaround for Windows as libvips keeps a reference to the input file
-    //        which makes it impossible to fs.unlink() it on cleanup stage
-    sharp.cache(false);
```
Comment (allouis) [inline core/server/lib/image/manipulator.js]: This is unnecessary as we handle file opening and closing - not libvips

File: core/server/web/shared/middlewares/image/handle-image-sizes.js [hunk 2]
```diff
@@ -51,12 +51,17 @@ module.exports = function (req, res, next) {
...
             })
             .then((resizedImageBuffer) => {
                 return storageInstance.saveRaw(resizedImageBuffer, req.url);
             });
     }).then(() => {
         next();
-    }).catch(next);
+    }).catch(function (err) {
+        if (err.code === 'SHARP_INSTALLATION') {
+            return redirectToOriginal();
```
Comment (allouis) [inline core/server/web/shared/middlewares/image/handle-image-sizes.js]: @ErisDS This redirects to the original image, if you try to get a resized image and sharp is not installed.

File: core/server/lib/image/manipulator.js (hunk 1 above)
Comment (kirrg001) [reply core/server/lib/image/manipulator.js]: ? this avoids that sharp caches the input file and keeps a reference to the file.

Comment (allouis) [reply core/server/lib/image/manipulator.js]: Sharp doesn't have a reference to the file, it just works on raw data as of https://github.com/TryGhost/Ghost/commit/28a686a936653fefc466f0f93b1210703d26fb43#diff-58eacecfe1e5c6bf4e29d13e7893b571 - we no longer pass in the file name as the input - but the raw data :relaxed:

Comment (kirrg001) [reply core/server/lib/image/manipulator.js]: Oh yeah, it was changed when we added the size comparison 👍 👻

File: core/server/lib/image/manipulator.js [hunk 3]
```diff
@@ -8,60 +8,20 @@ const fs = require('fs-extra');
...
 
+const unsafeProcess = (options = {}) => {
```
Comment (gargol) [inline core/server/lib/image/manipulator.js]: :heart: how tiny the method has become!

Comment (gargol) [reply core/server/lib/image/manipulator.js]: the ` withoutEnlargement: true` reduced it by one `if` and comparing two buffers in one method saved another bit. nice!

**Review by naz — APPROVED** [review]
This is great :ship: !

File: core/server/web/shared/middlewares/image/handle-image-sizes.js (hunk 2 above)
Comment (ErisDS) [reply core/server/web/shared/middlewares/image/handle-image-sizes.js]: 💃

File: core/server/lib/image/manipulator.js (hunk 3 above)
Comment (allouis) [reply core/server/lib/image/manipulator.js]: :grinning: glad you like!!!

Comment (kirrg001) [reply core/server/lib/image/manipulator.js]: I LIKE TOO 💥

**Pull request merged**